packages/backend/src/models/ProjectModel.ts

```diff
@@ -54,48 +57,148 @@ export class ProjectModel {
+    private async upsertWarehouseConnection(
+        trx: Transaction,
+        projectId: number,
+        data: CreateWarehouseCredentials,
     ): Promise<void> {
-        const projects = await this.database('projects')
-            .where('project_uuid', projectUuid)
-            .select('*');
-        if (projects.length === 0) {
-            throw new NotExistsError(
-                `No project exists with id ${projectUuid}`,
-            );
-        }
-        const [project] = projects;
         let encryptedCredentials: Buffer;
         try {
             encryptedCredentials = this.encryptionService.encrypt(
-                JSON.stringify(credentials),
+                JSON.stringify(data),
             );
         } catch (e) {
             throw new UnexpectedServerError('Could not save credentials.');
         }
-        await this.database('warehouse_credentials')
+        await trx('warehouse_credentials')
             .insert({
-                project_id: project.project_id,
-                warehouse_type: credentials.type,
+                project_id: projectId,
+                warehouse_type: data.type,
                 encrypted_credentials: encryptedCredentials,
             })
             .onConflict('project_id')
             .merge();
     }
 
+    async hasProjects(): Promise<boolean> {
+        const projects = await this.database('projects').select('project_uuid');
+        return projects.length > 0;
+    }
+
+    async create(
+        organizationUuid: string,
+        data: CreateProject,
+    ): Promise<string> {
+        const orgs = await this.database('organizations')
+            .where('organization_uuid', organizationUuid)
+            .select('*');
+        if (orgs.length === 0) {
+            throw new NotExistsError('Cannot find organization');
+        }
+        return this.database.transaction(async (trx) => {
+            try {
+                let encryptedCredentials: Buffer;
+                try {
+                    encryptedCredentials = this.encryptionService.encrypt(
+                        JSON.stringify(data.dbtConnection),
+                    );
+                } catch (e) {
+                    throw new UnexpectedServerError(
+                        'Could not save credentials.',
+                    );
+                }
+                const results = await trx('projects')
+                    .insert({
+                        name: data.name,
+                        organization_id: orgs[0].organization_id,
+                        dbt_connection_type: data.dbtConnection.type,
+                        dbt_connection: encryptedCredentials,
+                    })
+                    .onConflict('project_id')
+                    .merge()
+                    .returning('*');
+                if (results.length === 0) {
+                    throw new UnexpectedServerError(
+                        'Could not create project.',
+                    );
+                }
```

Comment:
Is this actually possible? I'm not sure it can happen

Reply:
correct, it shouldn't be possible, unless something weird happens in knex insert.